polartools/diffraction.py

```diff
@@ -0,0 +1,189 @@
+        plt.plot(xdata, fit.best_fit)
+        plt.show()
+    return fit
+
+
+def fit_series(
+    db,
+    scan_series,
+    model="Gaussian",
+    output=False,
+    var_series=None,
+    positioner="4C Theta",
+    detector="APD",
+    monitor=None,
+    **kwargs,
+):
+    """
+    Fit series of scan with reflection profile of choice and provide fit parameters as list.
+
+    Uses lmfit (https://lmfit.github.io/lmfit-py/).
+
+
+    Parameters
+    ----------
+    db : database
+        Databroker database. If None, it will attempt to read from csv or spec
+        files.
+    scan_series : int
+        start, stop, step, [start2, stop2, step2, ... ,startn, stopn, stepn]
+    model:
+        fit model: Gaussian, Lorentian, Voigt, PseidoVoigt
+    output:
+        Output fit parameters and plot data+fit for each scan.
+    var_series:
+        Varying variable for scan series, e.g. SampK (sample temperature), optional.
+    positioner : string
+        Name of the positioner, this needs to be the same as defined in
+        Bluesky.
+    detector : string
+        Detector to be read, needs to be the same name as in Bluesky.
+    monitor : string, optional
+        Name of the monitor detector. The returned scans will be
+        detectors/monitor.
+    kwargs :
+        Passed to `load_table` and 'fit_peak'.
+
+    Returns
+    -------
+    fit : lmfit ModelResult class
+        Contains the fit results. See:
+        https://lmfit.github.io/lmfit-py/model.html#the-modelresult-class
+    """
+
+    if len(scan_series) % 3:
+        raise ValueError(
+            f"expected 3*n={3*(len(scan_series)//3)} arguments, got {len(scan_series)}"
+        )
+
+    nbp = 0
+    for series in range(1, len(scan_series), 3):
+        nbp = (
+            nbp
+            + int(scan_series[series] - scan_series[series - 1])
+            / scan_series[series + 1]
+            + 1
+        )
+    fit_result = [np.zeros(7)]
+    for i in range(int(nbp) - 1):
+        fit_result.append(np.zeros(7))
+
+    index = 0
+    for series in range(1, len(scan_series), 3):
+        start = scan_series[series - 1]
+        stop = scan_series[series]
+        step = scan_series[series + 1]
+        print("Intervals: {} to {} with step {}".format(start, stop, step))
+        # fitnr=0
+        for scan in range(start, stop + 1, step):
+
+            table = load_table(
+                scan, db, detector=detector, monitor=monitor, **kwargs
+            )
+            data = [np.array(table[positioner])]
+            if monitor is None:
+                data.append(np.array(table[detector]))
+            else:
+                data.append(
+                    np.array(table[detector]) / np.array(table[monitor])
+                )
+            try:
+                data.append(np.array(table[var_series]))
+                fit_result[index][0] = data[2].mean()
+                print(f"{var_series} = {fit_result[index][0]}")
+            except:
```

Comment:
Another Codacy complain: this bare `except` will collect any error that might occur here, which can lead to errors elsewhere. I believe this is catching the situation where `var_series` is not in `table`, is that correct? If it is, you could do:
```
except KeyError:
```

Reply:
not needed anymore